test/gtest/CMakeLists.txt

```diff
@@ -29,24 +23,25 @@ function(add_gtest TEST_NAME)
       target_include_directories(test_${TEST_NAME} SYSTEM PRIVATE $<BUILD_INTERFACE:${FDEEP_INCLUDE_DIR}>)
       target_include_directories(test_${TEST_NAME} SYSTEM PRIVATE $<BUILD_INTERFACE:${EIGEN_INCLUDE_DIR}/eigen3>)
     endif()
+    target_link_libraries(test_${TEST_NAME} gtest gtest_main MIOpen ${Boost_LIBRARIES} hip::host $<BUILD_INTERFACE:roc::rocblas>)
     if(NOT MIOPEN_EMBED_DB STREQUAL "")
-      target_link_libraries(test_${TEST_NAME} gtest_main MIOpen ${Boost_LIBRARIES} hip::host $<BUILD_INTERFACE:roc::rocblas> $<BUILD_INTERFACE:miopen_data>)
-    else()
-      target_link_libraries(test_${TEST_NAME} gtest_main MIOpen ${Boost_LIBRARIES} hip::host $<BUILD_INTERFACE:roc::rocblas>)
+        target_link_libraries(test_${TEST_NAME} $<BUILD_INTERFACE:miopen_data>)
     endif()
     # Enable CMake to discover the test binary
     gtest_discover_tests(test_${TEST_NAME} PROPERTIES ENVIRONMENT "MIOPEN_USER_DB_PATH=${CMAKE_CURRENT_BINARY_DIR};MIOPEN_TEST_ALL=${MIOPEN_TEST_ALL};MIOPEN_TEST_MLIR=${MIOPEN_TEST_MLIR};MIOPEN_TEST_COMPOSABLEKERNEL=${MIOPEN_TEST_COMPOSABLEKERNEL}")
 
   endif()
 endfunction()
 
-file(GLOB TESTS *.cpp)
-# Remove files that do not contain tests
-foreach(SOURCE ${SOURCES})
-    list(REMOVE_ITEM TESTS ${CMAKE_CURRENT_SOURCE_DIR}/${SOURCE})
-endforeach()
-
-foreach(TEST ${TESTS})
-    get_filename_component(BASE_NAME ${TEST} NAME_WE)
-    add_gtest(${BASE_NAME})
-endforeach()
+add_gtest(api_convbiasactiv)
+add_gtest(cba_infer)
+add_gtest(conv_api)
+add_gtest(conv_api_strided_tensors)
+add_gtest(log_test)
+add_gtest(log_test_neg)
+add_gtest(na_infer)
+add_gtest(solver_convasm3x3u)
+
+if(NOT MIOPEN_BACKEND_OPENCL)
+  add_gtest(dumpTensorTest)
+endif()
```

Comment:
I don't follow why the loop was removed, do you suggest for us to add each test explicitly ?

Reply:
Yes. The `file(GLOBE...)` command was designed for CMake in _script mode_ (`*.cmake` files). Please read more about it here: https://cmake.org/cmake/help/latest/manual/cmake-language.7.html. The author's intention was not to use it in _build system generation mode_ - in this case. Please see the note for `GLOBE` here: https://cmake.org/cmake/help/latest/command/file.html#glob.
I will revert this change if you insist on returning the `file(GLOB...)` and the loop.

Reply:
@apwojcik This is not a big problem in this case. I think these changes should be reverted

Reply:
@apwojcik could you follow up with the review opinion above? Thanks!

Reply:
done